src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,179 @@
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { queryLanguages, attributesForPrediction = ['*'], nbHits = 1 } =
+      widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastAnswersResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRenderFn = debounce(renderFn, 200);
+
+    return {
+      $$type: 'ais.answers',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        // eslint-disable-next-line no-warning-comments
+        // FIXME: remove this customization once the engine accepts url encoded query params
+        if (instantSearchInstance.client.transporter) {
+          instantSearchInstance.client.transporter.userAgent.value =
+            'answers-test';
+        }
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance: initOptions.instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const { state, instantSearchInstance } = renderOptions;
+        const { query, index } = state;
+        const answersIndex = instantSearchInstance.client!.initIndex!(index);
+        if (!answersIndex.findAnswers) {
+          throw new Error(withUsage('`algoliasearch` >= 4.8.0 required.'));
+        }
+        if (!query) {
+          // renders nothing with empty query
+          lastAnswersResult = {};
+          isLoading = false;
+          renderFn(
+            {
+              ...this.getWidgetRenderState(renderOptions),
+              instantSearchInstance: renderOptions.instantSearchInstance,
+            },
+            false
+          );
+          return;
+        }
+
+        // render the loader
+        isLoading = true;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance: renderOptions.instantSearchInstance,
+          },
+          false
+        );
+
+        // call /answers API
+        runConcurrentSafePromise(
+          answersIndex.findAnswers(query, queryLanguages, {
+            nbHits,
+            attributesForPrediction,
+          })
+        ).then(result => {
+          lastAnswersResult = result;
+          lastAnswersResult.hits = addAbsolutePosition<typeof result.hits[0]>(
```

Comment:
what about queryID and __escaped?

Reply:
https://github.com/algolia/instantsearch.js/pull/4581/commits/87d96f7f8f08a4cf72da02b193bf10ca1c8f2ee3